tests/test_lib_resolver.py

```diff
@@ -1066,6 +1066,161 @@ def test_14_add_user_update_delete_objectGUID(self):
         self.assertFalse(user_id)
 
     @ldap3mock.activate
+    def test_15_invalid_additional_attributes(self):
+        ldap3mock.setLDAPDirectory(LDAPDirectory)
+        y = LDAPResolver()
+        classes = "top, inetOrgPerson"
+        additional_attribs = "invalid"
+        y.loadConfig({'LDAPURI': 'ldap://localhost',
+                      'LDAPBASE': 'o=test',
+                      'BINDDN': 'cn=manager,ou=example,o=test',
+                      'BINDPW': 'ldaptest',
+                      'LOGINNAMEATTRIBUTE': 'cn',
+                      'LDAPSEARCHFILTER': '(cn=*)',
+                      'LDAPFILTER': '(&(cn=%s))',
+                      'USERINFO': '{ "username": "cn",'
+                                  '"phone" : "telephoneNumber", '
+                                  '"mobile" : "mobile",'
+                                  '"email" : "email",'
+                                  '"password" : "userPassword",'
+                                  '"surname" : "sn", '
+                                  '"givenname" : "givenName", '
+                                  '"additionalAttr": "uidNumber" }',
+                      'ADDITIONAL_ATTRIBUTES': additional_attribs,
+                      'OBJECT_CLASSES': classes,
+                      'DN_TEMPLATE': "cn=<username>,ou=example,o=test",
+                      'UIDTYPE': 'DN',
+                      'NOREFERRALS': True
+        })
+
+        user = "achmed"
+
+        # First we add the user with add_user
+        r = y.add_user({"username" : user,
+                        "surname" : "Ali",
+                        "email" : "achmed.ali@example.com",
+                        "password" : "testing123",
+                        'mobile': ["1234", "45678"],
+                        "givenname" : "Achmed"})
+        self.assertTrue(r)
+
+        # Check that the additional attributes are available
+        user_id = y.getUserId("achmed")
+
+        # Now we delete the user with add_user
+        y.delete_user(user_id)
+        # Now there should be no achmed anymore
+        user_id = y.getUserId("achmed")
+        self.assertFalse(user_id)
+
+    @ldap3mock.activate
+    def test_16_add_user_update_delete_additional_attributes(self):
+        ldap3mock.setLDAPDirectory(LDAPDirectory)
+        y = LDAPResolver()
+        classes = "top, inetOrgPerson"
+        additional_attribs = { 'homeDirectory': '/home/achmed',
```

Comment:
I am not quite convince myself at the moment.
(maybe we should start an issue to discuss this).

You can however at the moment add arbitrary keys and values to the user infos. They will be added to the userlist. You will see these fields in the API but not in the  Web UI.

There might be two different types of userfields for *user_adding*:

**Fixed** values, like "every user will get the attribute *memberOf=groupA* and

**Editable** values, like the admin needs to enter an additional value, when adding  a user.

What are you after? The 2nd one (ediable) might get complicated. This needs to be reflected in the Web UI...